src/ldclient_backoff.erl

```diff
@@ -35,14 +36,21 @@
 
 -spec init(Initial :: non_neg_integer(), Max :: non_neg_integer(), Destination :: pid(), Value :: term()) -> backoff().
 init(Initial, Max, Destination, Value) ->
+    SafeInitial = lists:max([Initial, 1]),
     #{
-        initial => Initial,
-        current => Initial,
+        initial => SafeInitial, %% Do not allow initial delay to be 0 or negative.
+        current => SafeInitial,
         max => Max,
         attempt => 0,
         active_since => undefined,
         destination => Destination,
-        value => Value
+        value => Value,
+        %% The exponent at which the backoff delay will exceed the maximum.
+        %% Beyond this limit the backoff can be set to the max.
+        max_exp => math:ceil(math:log2(Max/SafeInitial))
+        %% For reasonable values this should ensure we never overflow.
```

Comment:
This does mean that you can set some initial reconnect that is vastly too large, but I don't think attempting to fix that has much value.